### DISCUSSION TITLE
is there utility for a `set_error_response` + `throw :rodauth_error` method?

### DISCUSSION THREAD

**Discussion by jf** [jeremyevans/rodauth#412] [General]
(To help make discussion easier, here is the code for all 3 methods, extracted out for easier viewing:
```
    def throw_error_status(status, field, error)
      set_response_error_status(status)
      throw_error(field, error)
    end

    def set_response_error_status(status)
      response.status = status
    end

    def throw_error(field, error)
      set_field_error(field, error)
      throw :rodauth_error
    end
```
)

I have been working on coding my own custom validation logic in a `before_create_account` block. I started with the sample code at http://rodauth.jeremyevans.net/rdoc/files/doc/guides/registration_field_rdoc.html (both examples use `throw_error_status(status, field, error)`, which I see basically adds a `throw_error(field, error)` call vs the code for `set_response_error_status`.

Because I have my own custom validation logic, I actually do my own `set_field_error` errors... and because I actually validate **all** of the fields at once, I am not doing `throw_error_status(status, field, error)`... which would only `set_field_error` for *one* field, before throwing.

My solution has been to basically do this at the end after all the validation (assuming errors were encountered):
```
response.status = 422
throw :rodauth_error
```

It appears that the throw is essential based on testing... or else account creation would carry on. So my question is, is there utility for a `set_error_response` + `throw :rodauth_error` method? I do not see anything like that in the methods provided.

**Comment by jeremyevans** [comment]
Can you find places in Rodauth's code where it would use such a method?  If not, I don't think Rodauth should ship with such a method.  It's easy to add yourself, either directly or as a custom feature, should you need it.

**Reply by jf** [reply]
for folks in my situation, it would be helpful. Not sure about Rodauth itself though.

If I were to add it myself, could I do the `def` inside my `plugin :rodauth` block?

**Reply by jeremyevans** [reply]
The plugin block is in configuration scope, not auth scope. You have to wrap the `def` in an `auth_class_eval` block.  See https://github.com/jeremyevans/rodauth/blob/master/spec/rodauth_spec.rb#L697 for an example.

**Reply by jf** [reply]
actually a thought: what about if `throw_error_status` were to be modified so that the `field` and `error` params were optional? would you accept a patch for that? to be clear, `throw :rodauth_error` will still happen; the only question is whether `set_field_error` is called

**Reply by jeremyevans** [reply]
I don't think the benefits of making the parameters optional is worth the cost.  If you want a separate method added that does `throw :rodauth_error`, I suppose that could be added.

**Reply by jf** [reply]
thank you. I will work on a PR soon

(Update: PR [is here](https://github.com/jeremyevans/rodauth/pull/418))
